packages/hdwallet-ledger-webusb/src/transport.ts

```diff
@@ -86,18 +95,76 @@ export class LedgerWebUsbTransport extends LedgerTransport {
   }
 
   public async call(coin: string, method: string, ...args: any[]): Promise<LedgerResponse> {
+    return  await this.sendToLedger({
+      fromCallFn: true,
+      args,
+      method,
+      coin
+    })
+  }
+
+  public async sendToLedger(sendArgs: LedgerRequest): Promise<LedgerResponse> {
+    let { action, coin, method, args, fromCallFn } = sendArgs
     let response
 
-    try {
-      this.emit(`ledger.${coin}.${method}.call`, makeEvent({
-        message_type: method,
-        from_wallet: false,
-        message: {}
-      }))
+    // set it back to false in case it was true
+    if (this.isCancelled) this.isCancelled = false
 
-      response = await new (translateCoin(coin))(this.transport)[method](...args)
+    try {
+      await retry(async (bail) => {
+        if (this.isCancelled) {
+          this.isCancelled = false
+          bail()
+          return
+        }
+
+        // open transport if it's closed
+        await this.open()
+
+        response = fromCallFn ?
+          await new (translateCoin(coin))(this.transport)[method](...args) :
+          await action(this.transport)
+
+        // close transport after every call
+        await this.close()
+
+        this.emitEvent({
+          coin: 'none',
+          method: 'retry',
+          response,
+          fromWallet: true,
+          eventType: 'success'
+        })
+      }, {
+        retries: 20,
```

Comment:
what does this do when it hits `20`th retry?

Reply:
After the 20th retry, it will throw the last error encountered. This is just an arbitrary number and can be set to whatever makes sense. It can also be a value that is passed. Not really sure what makes sense until I can test it via axiom

Reply:
cool. let's pull that up in to a SCREAMING_CONSTANT up near the top of the file.

Reply:
lol sounds good